### DISCUSSION TITLE
How to combine multiple 'uses' in github actions?

### DISCUSSION THREAD

**Discussion by aahnik** [community/community#26579] [Actions]
<p>I find myself repeating these three lines.</p>
<pre><code class="lang-yaml">    steps:
      - uses: actions/checkout@v2
      - uses: actions/setup-python@v2
      - uses: aahnik/setup-pyproject@main
</code></pre>
<p>How can I combine these three ?</p>
<p>In the composite action, “uses” is not allowed…</p>

**Comment by mickeygousset** [comment, marked as answer]
<p>You can’t.  A step can only run one action or task at a time.  So you have to list them out separately.</p>

**Comment by fregante** [comment]
<p>If GitHub Actions supports the fully YAML specification, try Anchors:</p>
<p><a href="https://medium.com/@kinghuang/docker-compose-anchors-aliases-extensions-a1e4105d70bd" class="onebox" target="_blank" rel="noopener nofollow ugc">https://medium.com/@kinghuang/docker-compose-anchors-aliases-extensions-a1e4105d70bd</a></p>
<aside class="onebox stackexchange">
  <header class="source">
      <a href="https://stackoverflow.com/questions/48940619/yaml-how-to-reuse-single-string-content" target="_blank" rel="noopener nofollow ugc">stackoverflow.com</a>
  </header>
  <article class="onebox-body">
      <a href="https://stackoverflow.com/users/503831/arnold-roa" target="_blank" rel="noopener nofollow ugc">
    <img alt="Arnold Roa" src="https://user-images.githubusercontent.com/1402241/181098555-14c8e9b6-105d-4b9a-b03f-bfed27be4e0e.jpeg" class="thumbnail onebox-avatar" width="128" height="128">
  </a>
<h4>
  <a href="https://stackoverflow.com/questions/48940619/yaml-how-to-reuse-single-string-content" target="_blank" rel="noopener nofollow ugc">YAML how to reuse single string content</a>
</h4>

<div class="tags">
  <strong>yaml</strong>
</div>

<div class="date">
  asked by
  
  <a href="https://stackoverflow.com/users/503831/arnold-roa" target="_blank" rel="noopener nofollow ugc">
    Arnold Roa
  </a>
  on <a href="https://stackoverflow.com/questions/48940619/yaml-how-to-reuse-single-string-content" target="_blank" rel="noopener nofollow ugc">03:23AM - 23 Feb 18 UTC</a>
</div>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>I suppose you could specify the list once at the top and then recall it in every job <em>in the same file</em></p>

**Comment by mickeygousset** [comment]
<p>Right now the GitHub Actions YAML doesn’t support anchors.</p>
